packages/form-js-viewer/src/render/components/FormField.js

```diff
@@ -61,35 +64,47 @@ export default function FormField(props) {
 
   const onBlur = useCallback(() => {
     if (viewerCommands) {
-      viewerCommands.updateFieldValidation(field, value);
+      viewerCommands.updateFieldValidation(field, value, indexes);
     }
     eventBus.fire('formField.blur', { formField: field });
-  }, [ eventBus, viewerCommands, field, value ]);
+  }, [ eventBus, viewerCommands, field, value, indexes ]);
 
   const onFocus = useCallback(() => {
     eventBus.fire('formField.focus', { formField: field });
   }, [ eventBus, field ]);
 
   useEffect(() => {
     if (viewerCommands && initialValue) {
-      viewerCommands.updateFieldValidation(field, initialValue);
+      viewerCommands.updateFieldValidation(field, initialValue, indexes);
     }
-  }, [ viewerCommands, field, initialValue ]);
+  }, [ viewerCommands, field, initialValue, JSON.stringify(indexes) ]);
 
   const hidden = useCondition(field.conditional && field.conditional.hide || null);
 
+  const onChangeIndexed = useCallback((update) => {
+
+    // add indexes of the keyed field to the update, if any
+    onChange(FormFieldComponent.config.keyed ? { ...update, indexes } : update);
+  }, [ onChange, FormFieldComponent.config.keyed, indexes ]);
+
   if (hidden) {
-    return <Empty />;
+    return <Hidden field={ field } />;
   }
 
+  const domId = `${prefixId(field.id, formId, indexes)}`;
+  const fieldErrors = get(errors, [ field.id, ...Object.values(indexes || {}) ]) || [];
+  const errorMessageId = errors.length === 0 ? undefined : `${domId}-error-message`;
```

Comment:
We should open an [issue in our examples](https://github.com/bpmn-io/form-js-examples/tree/master/custom-components) to use these new props (ease up things a lot).

Reply:
https://github.com/bpmn-io/form-js-examples/issues/9